### PR TITLE
wip: add test for valid markup before save

### PR DESCRIPTION
**Pull request by PaulBunker**
### Problem
Users can add bad data (or even just be in the process of typing good data) and this causes the Gatsby dev server to throw errors and omit pages from the build.

 - Validate mdx before saving to disk to avoid build problems.

UPDATE: I've found a way to check if the form field will cause gatsby to throw errors. Not sure how to go about adding an error state to the form and disabling the save button

### CODE REVIEW

**Comment by PaulBunker** [comment]
Next figure out how to set the error state on the form field if we are in an error state

File: packages/gatsby-tinacms-mdx/src/useMdxForm.tsx [hunk 1]
```diff
@@ -161,8 +163,33 @@ export function useMdxForm(
...
 
+  const isValidMdxState = (mdxContent: any) => {
+    const wrapped = `<div>${mdxContent}</div>`
+    try {
+      transform(wrapped, {
+        objectAssign: 'Object.assign',
+      })
+      return true
+    } catch (err) {
+      console.log('🚨 Not saving until your mdx is vallid')
```
Comment (DirtyF) [inline packages/gatsby-tinacms-mdx/src/useMdxForm.tsx]:
```suggestion
      console.log('🚨 Not saving until your MDX is valld')
```

**Comment by PaulBunker** [comment]
For now I will just use

```
 {
      label: "Body",
      name: "rawMarkdownBody",
      component: "textarea",
      validate: function(value) {
        const wrapped = `<div>${value}</div>`
        try {
          transform(wrapped, {
            objectAssign: "Object.assign",
          })
        } catch (err) {
          return "🚨 Malformed MDX please fix before save"
        }
      },
    },
```

In the page component where I define my tina form fields. It's not ideal because I'm not setting the state of the form on't the field but it seems to work better than doing the check in the plugin at the moment. I'll have another crack asap but I think I need to look a little more into Final Form.

**Comment by PaulBunker** [comment]
image

**Comment by PaulBunker** [comment]
kindafix

**Comment by ncphillips** [comment]
Why did you decide to close this @PaulBunker ?

**Comment by PaulBunker** [comment]
I just thought it looked messy. I've not given up just struggling a little.

**Pull request closed without merging**